src/watertap_contrib/seto/analysis/net_metering/PV_RO.py

```diff
@@ -0,0 +1,457 @@
+
+from pyomo.environ import (
+    ConcreteModel,
+    Objective,
+    Param,
+    Constraint,
+    Var,
+    Block,
+    TransformationFactory,
+    assert_optimal_termination,
+    value,
+    units as pyunits,
+)
+from pyomo.network import Arc
+from idaes.core import FlowsheetBlock
+from idaes.core.solvers.get_solver import get_solver
+from idaes.models.unit_models import Product, Feed
+from idaes.core.util.model_statistics import *
+from idaes.core.util.scaling import *
+from idaes.core import UnitModelCostingBlock
+from idaes.core.util.initialization import propagate_state
+from idaes.apps.grid_integration.multiperiod.multiperiod import MultiPeriodModel
+
+from watertap.core.util.initialization import assert_degrees_of_freedom
+from watertap.property_models.NaCl_prop_pack import NaClParameterBlock
+from watertap.unit_models.pressure_changer import Pump, EnergyRecoveryDevice
+
+from watertap.unit_models.reverse_osmosis_0D import (
+    ReverseOsmosis0D,
+    ConcentrationPolarizationType,
+    MassTransferCoefficient,
+    PressureChangeType,
+)
+from watertap.examples.flowsheets.RO_with_energy_recovery.RO_with_energy_recovery import (
+    calculate_operating_pressure,
+)
+from watertap.core.util.infeasible import *
+
+from watertap_contrib.seto.analysis.net_metering.util import (
+    display_ro_pv_results,
+    display_pv_results,
+)
+from watertap_contrib.seto.costing import (
+    TreatmentCosting,
+    EnergyCosting,
+    SETOSystemCosting,
+)
+from watertap_contrib.seto.solar_models.zero_order import PhotovoltaicZO
+from watertap_contrib.seto.core import SETODatabase, PySAMWaterTAP
+
+
+solver = get_solver()
+
+absolute_path = os.path.dirname(__file__)
+print(absolute_path)
+
+tech_config_file = "/pysam_data/pvsamv1.json"
+tech_config_file = absolute_path + tech_config_file
+grid_config_file = "/pysam_data/grid.json"
+grid_config_file = absolute_path + grid_config_file
+rate_config_file = "/pysam_data/utilityrate5.json"
+rate_config_file = absolute_path + rate_config_file
+cash_config_file = "/pysam_data/singleowner.json"
+cash_config_file = absolute_path + cash_config_file
+weather_file = "/pysam_data/phoenix_az_33.450495_-111.983688_psmv3_60_tmy.csv"
+weather_file = absolute_path + weather_file
+
+
+def build_ro_pv():
+    """Builds the structure of the PV-RO system
+
+    Returns:
+        object: A Pyomo concrete optimization model and flowsheet
+    """
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(dynamic=False)
+    m.db = SETODatabase()
+    m.pysam = PySAMWaterTAP(
+        pysam_model="pv",
+        tech_config_file=tech_config_file,
+        grid_config_file=grid_config_file,
+        rate_config_file=rate_config_file,
+        cash_config_file=cash_config_file,
+        weather_file=weather_file,
+    )
+    m.fs.properties = NaClParameterBlock()
+
+    treatment = m.fs.treatment = Block()
+    energy = m.fs.energy = Block()
+
+    energy.pv = PhotovoltaicZO(property_package=m.fs.properties, database=m.db)
```

Comment:
I think we should change this into a regular unit model (as opposed to ZO model) and just exclude the property package related config options. However, for the sake of moving things along, we can reserve such changes for a subsequent PR.

Reply:
Yes agreed- this implementation was meant to be temporary but caught legs and here we are. I will address this and might as well do it before we get the test file in so we don't have to redo that as well.